misc/services/mobile-webui/mobile-webui-frontend/src/reducers/wfProcesses_status/picking.js

```diff
@@ -91,6 +91,8 @@ export const generateAlternativeSteps = ({ draftDataStored, lineId, stepId, qtyT
 
       const qtyToAllocateThisStep = Math.min(qtyToAllocateRemaining, qtyAvailableToAllocateInThisStep);
 
+      if (qtyToAllocateThisStep === 0) break;
```

Comment:
Pls add unit tests for each case you discover.
That would be MANDATORY in order to maintain this algorithm.